### PR TITLE
Features/web support,settings,devices,persistency,auto sync

### PR DESCRIPTION
**Pull request by JacobRoe**
See commit descriptions and related tickets for more information.

Related to:
- [OPTM-42]
- [OPTM-43]
- [OPTM-44]
- [OPTM-45]
- [OPTM-46]

[OPTM-42]: https://ods-p2p-task-mgmt.atlassian.net/browse/OPTM-42
[OPTM-43]: https://ods-p2p-task-mgmt.atlassian.net/browse/OPTM-43
[OPTM-44]: https://ods-p2p-task-mgmt.atlassian.net/browse/OPTM-44
[OPTM-45]: https://ods-p2p-task-mgmt.atlassian.net/browse/OPTM-45
[OPTM-46]: https://ods-p2p-task-mgmt.atlassian.net/browse/OPTM-46

### CODE REVIEW

**Comment by guisedlythe** [comment]
I fixed the test and merged main back into this branch. It should be read for merge now.
We will continue the refactoring in more fine-grained feature branches from now on.

**Review by firstdayofjune — APPROVED** [review]
Only have some minor comments on code readability. Since this PR is a blocker I will still merge it into main, so we unblock the further development.
I commented all the changes I deem necessary to OPTM-49 so we don't forget them after the merge.

**Pull request merged**
